src/renderer/components/NewProjectModal/NewProjectView.tsx

```diff
@@ -45,30 +45,57 @@ const NewProjectView = ({
   setProjectName,
 }: Props) => {
   const dispatch = useDispatch();
+  const [isAwaitingProjectName, setIsAwaitingProjectName] =
+    useState<boolean>(true);
 
-  const setProjectInStore = async (project: RuntimeProject) => {
-    dispatch(projectCreated(project));
-  };
+  const setProjectInStore = useCallback(
+    async (project: RuntimeProject) => {
+      dispatch(projectCreated(project));
+    },
+    [dispatch]
+  );
 
-  const handleContinue = async () => {
+  const handleContinue = useCallback(async () => {
     const project = await makeProjectWithoutMedia(projectName);
     if (project === null) {
       return;
     }
     setProjectInStore(project);
     nextView();
-  };
+  }, [nextView, projectName, setProjectInStore]);
+
+  useEffect(() => {
```

Comment:
This code is duplicated between the two pages - can we bring it up a level? Or maybe refactor into a custom hook